core/controllers/reader.py

```diff
@@ -402,9 +404,25 @@ def post(self, exploration_id):
             self.payload.get('params'),
             feconf.PLAY_TYPE_NORMAL)
 
+        if user_id:
+            learner_progress_services.add_exp_id_to_completed_list(
+                user_id, exploration_id)
+
         if user_id and collection_id:
             collection_services.record_played_exploration_in_collection_context(
                 user_id, collection_id, exploration_id)
+            collections_left_to_complete = (
+                collection_services.get_next_exploration_ids_to_complete_by_user( # pylint: disable=line-too-long
+                    user_id, collection_id))
+
+            if not collections_left_to_complete:
+                learner_progress_services.add_collection_id_to_completed_list(
```

Comment:
Ah, your design doc addresses this. Ignore this comment (though you might want to consider the above suggestion and evaluate it against what you currently have).